modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/jms/client/Send.java

```diff
@@ -90,18 +82,22 @@ public BValue execute(Context context) {
             throw new BallerinaException("Need to use a JMSConnector as the first argument", context);
         }
         //Getting ballerina message and extract carbon message.
-        BMessage bMessage = (BMessage) getArgument(context, 5);
+        BMessage bMessage = (BMessage) getArgument(context, 3);
         if (bMessage == null) {
             throw new BallerinaException("Ballerina message not found", context);
         }
         CarbonMessage message = bMessage.value();
         //Create property map to send to transport.
         Map<String, String> propertyMap = new HashMap<>();
         //Getting the map of properties.
-        BMap properties = (BMap) getArgument(context, 6);
+        BMap properties = ((ClientConnector) connector).getProperties();
+
+        for (BString key : (Set<BString>) properties.keySet()) {
```

Comment:
The variable 'properties' in this case is a BMap, which doesn't provide a way to get entry set. Anyway will replace this with stream.collect.